tests/bluetooth/tester/src/btp_ots.c

```diff
@@ -0,0 +1,336 @@
+/* btp_ots.c - Bluetooth OTS Tester */
+
+/*
+ * Copyright (c) 2024 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/bluetooth/services/ots.h>
+
+#include <zephyr/sys/byteorder.h>
+#include <stdint.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME bttester_ots
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_BTTESTER_LOG_LEVEL);
+
+#include "btp/btp.h"
+
+#define OBJ_POOL_SIZE CONFIG_BT_OTS_MAX_OBJ_CNT
+#define OBJ_MAX_SIZE  100
+
+static struct object {
+	uint8_t data[OBJ_MAX_SIZE];
+	char name[CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1];
+	bool in_use;
+} objects[OBJ_POOL_SIZE];
+
+struct object_creation_data {
+	struct object *object;
+	struct bt_ots_obj_size size;
+	uint32_t props;
+};
+
+#define OTS_OBJ_ID_TO_OBJ_IDX(id) (((id) - BT_OTS_OBJ_ID_MIN) % ARRAY_SIZE(objects))
+
+static struct object_creation_data *object_being_created;
+
+static struct bt_ots *ots;
+
+static uint8_t ots_supported_commands(const void *cmd, uint16_t cmd_len,
+				      void *rsp, uint16_t *rsp_len)
+{
+	struct btp_ots_read_supported_commands_rp *rp = rsp;
+
+	tester_set_bit(rp->data, BTP_OTS_READ_SUPPORTED_COMMANDS);
+	tester_set_bit(rp->data, BTP_OTS_REGISTER_OBJECT);
+
+	*rsp_len = sizeof(*rp) + 1;
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static struct object *get_object(void)
+{
+	for (unsigned int i = 0; i < ARRAY_SIZE(objects); i++) {
```

Comment:
```suggestion
	for (size_t i = 0; i < ARRAY_SIZE(objects); i++) {
```

`ARRAY_SIZE` return a `size_t`